lib/rspec/mocks/method_double.rb

```diff
@@ -249,6 +249,19 @@ def new_rspec_prepended_module
         end
       end
 
+      def method_owner
+        @method_owner ||=
+          if Object === object
+            # This works in Ruby 2.3+, so we can just use this after we
+            # drop Ruby 2.2 and earlier.
```

Comment:
We've dropped Ruby 2.2 in the base `4-0-dev` branch.

Reply:
Maybe this will help to resolve [the conversation about `method_handle_for`](https://github.com/rspec/rspec-mocks/pull/1218#discussion_r180487361).

Reply:
FYI @godfat
you can remove the fallback logic for older rubies

Reply:
Thanks, I updated to use the one works in Ruby 2.3+